src/plugins/condition/components/inspector/StylesView.vue

```diff
@@ -0,0 +1,600 @@
+        getObjectsAndItemsFromSelection() {
+            let domainObject;
+            let subObjects = [];
+            let itemsWithConditionalStyles = 0;
+
+            //multiple selection
+            let itemInitialStyles = [];
+            let itemStyle;
+            this.selection.forEach((selectionItem) => {
+                const item = selectionItem[0].context.item;
+                const layoutItem = selectionItem[0].context.layoutItem;
+                const isChildItem = selectionItem.length > 1;
+                if (!isChildItem) {
+                    domainObject = item;
+                    itemStyle = getApplicableStylesForItem(item);
+                    if (this.hasConditionalStyle(item)) {
+                        itemsWithConditionalStyles += 1;
+                    }
+                } else {
+                    this.canHide = true;
+                    domainObject = selectionItem[1].context.item;
+                    if (item && !layoutItem || this.isItemType('subobject-view', layoutItem)) {
+                        subObjects.push(item);
+                        itemStyle = getApplicableStylesForItem(item);
+                        if (this.hasConditionalStyle(item)) {
+                            itemsWithConditionalStyles += 1;
+                        }
+                    } else {
+                        itemStyle = getApplicableStylesForItem(domainObject, layoutItem || item);
+                        this.items.push({
+                            id: layoutItem.id,
+                            applicableStyles: itemStyle
+                        });
+                        if (this.hasConditionalStyle(item, layoutItem)) {
+                            itemsWithConditionalStyles += 1;
+                        }
+                    }
+                }
+                itemInitialStyles.push(itemStyle);
+            });
+            this.isStaticAndConditionalStyles = this.isMultipleSelection && itemsWithConditionalStyles;
+            const {styles, mixedStyles} = getConsolidatedStyleValues(itemInitialStyles);
+            this.initialStyles = styles;
+            this.mixedStyles = mixedStyles;
+
+            this.domainObject = domainObject;
+            this.removeListeners();
+            if (this.domainObject) {
+                this.stopObserving = this.openmct.objects.observe(this.domainObject, '*', newDomainObject => this.domainObject = newDomainObject);
+                this.stopObservingItems = this.openmct.objects.observe(this.domainObject, 'configuration.items', this.updateDomainObjectItemStyles);
+            }
+
+            subObjects.forEach(this.registerListener);
+        },
+        updateDomainObjectItemStyles(newItems) {
+            //check that all items that have been styles still exist. Otherwise delete those styles
+            let keys = Object.keys(this.domainObject.configuration.objectStyles || {});
+            keys.forEach((key) => {
+                if ((key !== 'styles') &&
```

Comment:
Lets move this complex if statement into a descriptive helper function. If the helper function is descriptive enough, we can get rid of the comment.

Reply:
Done